### PR TITLE
feat: Add TypeScript definitions

### PR DESCRIPTION
**Pull request by tbjers**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/mtrezza/parse-server-api-mail-adapter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/mtrezza/parse-server-api-mail-adapter/issues?q=is%3Aissue).

### Issue Description

Related issue: #89
Closes: #89

### TODOs before merging

- [x] Add changes to documentation (guides, repository pages, in-code descriptions)

### CODE REVIEW

**Comment by mtrezza** [comment]
Never mind https://github.com/parse-community/parse-server-api-mail-adapter/pull/90#issuecomment-1756054270, this repository has the release branch as the working branch, hence the comment.

**Comment by tbjers** [comment]
I am not sure if there is a need to update the documentation with anything. Thoughts, @mtrezza?

**Comment by mtrezza** [comment]
Wouldn't the typescript definition files normally be committed with the repository? Or how are generated?

**Comment by tbjers** [comment]
They are automatically generated by the npm build command, so they would be generated at the same time as the rest of the code in lib.

**Comment by mtrezza** [comment]
Got it; regarding docs I don't think there's anything that needs to be added. This will be a major release without any breaking changes, correct?

**Comment by tbjers** [comment]
Correct.

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  // Change this to match your project
```
Comment (mtrezza) [inline tsconfig.json:2]: Could you remove these general comments, unless there is one specific to our implementation

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by mtrezza** [comment]
@sadortun could you review this PR please?

**Review by sadortun — APPROVED** [review]

**Comment by sadortun** [comment]
@mtrezza Yes, the file should be added to the repository.

**Comment by tbjers** [comment]
> @sadortun could you review this PR please? I wonder whether TS files should be committed to the repo as the IDE will need them?

cc: @sadortun The files do not need to be committed to the repository because they will be part of the package tarball as they are generated at the same time as everything else in `lib/`. My reasoning here is that we do not commit the generated code to the repository, either.

**Comment by sadortun** [comment]
@tbjers @mtrezza Ah, I was referring to `tsconfig.json` in this PR. And you're referring to `.d.ts` 😂

If `.d.ts` are generated by the build step, then you don't need them in the repo.

**Comment by mtrezza** [comment]
Thanks @sadortun for confirming, and thanks @tbjers for the PR; I'll go ahead and merge.

**Review by mtrezza — APPROVED** [review]

**Comment by parseplatformorg** [comment]
🎉 This pull request has been released in version [3.1.0](https://github.com/parse-community/parse-server-api-mail-adapter/releases/tag/3.1.0)

**Pull request merged**